src/Raven.Server/Documents/Sharding/ShardedDocumentDatabase.cs

```diff
@@ -52,8 +52,8 @@ protected override void InitializeCompareExchangeStorage()
 
     protected override void InitializeAndStartPeriodicDocumentsMigrator()
     {
-        PeriodicDocumentsMigrator = new ShardedPeriodicDocumentsMigrator(this);
-        PeriodicDocumentsMigrator.Start();
+        DocumentsMigrator = new ShardedDocumentsMigrator(this);
+        DocumentsMigrator.ExecuteMoveDocumentsAsync().Wait();
```

Comment:
Isn't that expensive? Why are you waiting for this and hold up the db startup?

Reply:
done